Godeps/_workspace/src/k8s.io/kubernetes/pkg/proxy/userspace/proxier.go

```diff
@@ -413,7 +435,12 @@ func (proxier *Proxier) OnServiceUpdate(services []api.Service) {
 			}
 
 			glog.V(1).Infof("Adding new service %q at %s:%d/%s", serviceName, serviceIP, servicePort.Port, servicePort.Protocol)
-			info, err = proxier.addServiceOnPort(serviceName, servicePort.Protocol, proxyPort, proxier.udpIdleTimeout)
+			svcRef, err := api.GetReference(service)
```

Comment:
GetReference is a really bad idea.  I'd just create an Object reference and pass it in to addServiceOnPort (as a value, not a pointer) and remove `serviceName`.  Set Kind, Namespace, Name, and UID.  Don't set anything else (unless you need it).